### PR TITLE
Containerising and automating the tools used for Bug triage and CI health reports

### PR DESCRIPTION
**Pull request by Girish5tri**
SUMMARY
Integrates the CI nightly run script into the ansible-network-triager tool and enhances its CLI functionality. This update centralizes configuration management and prepares the tool for future containerization and automation efforts. Key changes include adding new CLI commands, consolidating configuration into a single file, and implementing  email notifications and logging.

ISSUE TYPE
Enhancement

COMPONENT NAME
ansible-network-triager tool

CLI Commands:

Bug Scrub Command:
triager --bugs -c example-config.yaml --log --send-email
CI Report Command:
triager --ci -c example-config.yaml --log --send-email

To securely store sensitive information like email passwords and recipients, we'll use GitHub Secrets:
a) Go to your GitHub repository.
b) Click on "Settings" tab.
c) In the left sidebar, click on "Secrets and variables", then "Actions".
d) Click on "New repository secret".
e) Add the following secrets:

Name: EMAIL_SENDER ,  Value: [Your tools email address from which you want to send emails]
Name: EMAIL_PASSWORD, Value: [Your tools email password]
MAINTAINERS: A JSON string containing the maintainers information. 

For example:
[{"name": "John Doe", "email": "john@doe.localhost"}, {"name": "Jon Smith", "email": "jon@smith.localhost"}]

### CODE REVIEW

File: example-config.yaml [hunk 1]
```diff
@@ -10,19 +10,69 @@ orgs:
...
+      - name: ansible.security
+      - name: ansible.network
+      - name: frr.frr
+      - name: openvswitch.openvswitch
+      - name: community.yang
+      - name: ansible.utils
+      - name: ibm.qradar
+      - name: trendmicro.deepsec
+      - name: splunk.es
+    ci_repos:
```
Comment (KB-perByte) [inline example-config.yaml]: Please combine it to be a single config source, and make it generic enough to make the configuration not specific to ansible

Comment (Girish5tri) [reply example-config.yaml]: Sure will do it thanks

Comment (Girish5tri) [reply example-config.yaml]: Updated the tool to be more generic in the recent commit, can you please review the changes

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name: Triager Workflow
+
+on:
+  schedule:
```
Comment (KB-perByte) [inline .github/workflows/main.yml]: this need not be one single workflow, it can be two separate ones

Comment (Girish5tri) [reply .github/workflows/main.yml]: Sure will create two separate workflows for ci and bug scrub, thanks

Comment (Girish5tri) [reply .github/workflows/main.yml]: Done updated it in new commits

File: .github/workflows/ci_workflow.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: CI Report Workflow
```
Comment (KB-perByte) [inline .github/workflows/ci_workflow.yml]:
```suggestion
name: Nightly CI Report Workflow
```

File: example-config.yaml [hunk 4]
```diff
@@ -1,28 +1,97 @@
...
+#github_token: ${GITHUB_TOKEN}  #Use GITHUB_TOKEN if you have a GitHub Actions workflow configured, otherwise comment this line out for unauthenticated requests."
+
+maintainers: ${MAINTAINERS}
+
+# github_token: 'Manually enter your github token here to make authenticated requests or comment this line to make unauthenticated API requests'
+
+
+
+
+##Generic structure
```
Comment (KB-perByte) [inline example-config.yaml]: Strip this data out make example-config and add comments as explanation, 
Make this file as config.yaml should be in form of a template.

Comment (Girish5tri) [reply example-config.yaml]: Sure will do it thanks

Comment (Girish5tri) [reply example-config.yaml]: updated the changes in new commit

File: triager/config.py [hunk 5]
```diff
@@ -10,70 +11,117 @@
...
             cfg = "./config.yaml"
 
-        logging.info("attempting to read config file: {0}".format(cfg))
+        logging.info(f"attempting to read config file: {cfg}")
 
         try:
             with open(cfg, "r") as config_file:
-                config = yaml.safe_load(config_file)
-            logging.info("config file successfully loaded")
+                config_content = config_file.read()
```
Comment (KB-perByte) [inline triager/config.py:29]: Normalize these class operations and break them into smaller methods.

File: repo_config.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{
+    "ansible-collections": {
```
Comment (KB-perByte) [inline repo_config.json]: this file needs to be a part of the GH environments. we can expose an example schema!

File: README.md [hunk 7]
```diff
@@ -17,18 +11,118 @@ When run with `--send-email` it also emails this table to all the listed maintai
...
+
+
+3. Run the CI report:
+
+   docker-compose run triager --ci -c config.yaml --log --send-email
+
+
+### GitHub Actions
+
+1. Store your secrets (EMAIL_SENDER, EMAIL_PASSWORD, MAINTAINERS, GITHUB_TOKEN) in GitHub Actions secrets.
```
Comment (KB-perByte) [inline README.md]: https://github.com/ansible/ansible-content-actions take the Readme as an example to show the usage of GHA

File: .github/workflows/bug_triager_workflow.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
+name: Bug Triage Workflow
+
+on:
```
Comment (KB-perByte) [inline .github/workflows/bug_triager_workflow.yml:3]: Try adding pre-commit for some code specific checks 
https://github.com/ansible-collections/cisco.ios/blob/main/.pre-commit-config.yaml

**Review by KB-perByte — CHANGES_REQUESTED** [review]

File: .github/workflows/bug_triager_workflow.yml (hunk 8 above)
Comment (Girish5tri) [reply .github/workflows/bug_triager_workflow.yml:3]: sure will do it thanks

File: triager/config.py (hunk 5 above)
Comment (Girish5tri) [reply triager/config.py:29]: done updated in recent commit

File: repo_config.json (hunk 6 above)
Comment (Girish5tri) [reply repo_config.json]: Done. I've moved the configuration to GitHub environments.

File: README.md (hunk 7 above)
Comment (Girish5tri) [reply README.md]: Done, updated the Readme section

File: .github/workflows/bug_triager_workflow.yml (hunk 8 above)
Comment (Girish5tri) [reply .github/workflows/bug_triager_workflow.yml:3]: Completed

**Review by KB-perByte — APPROVED** [review]

**Pull request merged**